### DISCUSSION TITLE
Changing the Throughput starting status

### DISCUSSION THREAD

**Discussion by manojshah1976** [nbrown02/FlowViz-Jira#20] [Q&A] [answered]
Is there a way to change the status that drives the Through Put currently its based on the ticket creation date but would like to trigger this when the ticket has been set to our internal Development status, is this possible?

**Comment by nbrown02** [comment, marked as answer]
You would have to use one of the queries that brings back all the work item history then take the timestamp of when an item first hits that particular "status". This isn't something I will be looking to create but could share brief (free) guidance on how to go about it.

**Reply by manojshah1976** [reply]
Totally understand I guess this is a sort of unique need but what we find is that when the items are initially set to a ToDo status they need further refinement and whilst that would work in most circumstances its not quite right for us. Appreciate a high level view and happy to share any progress on this front.

**Reply by nbrown02** [reply]
You'll need to have a table with all the changes to an issue (you can potentially just use one of those that already exists) as you need the specific timestamp when it went into the status you are after. 

Here is what it would look like using the WorkItems History table (and a fictional status of 'Dev Done'):

- You'll need to add two calculated columns, the first being: 
CDate = IF('WorkItems History'[To] = "Dev Done", 'WorkItems History'[ChangedDate])

- The second being:
CompletedDate = CALCULATE (
    MIN ( 'WorkItems History'[CDate] ),
    ALLEXCEPT ( 'WorkItems History', 'WorkItems History'[IssueId] )
)

The second will give you the first time it went into that status, if you wanted the most recent then replace MIN with MAX in the second calculated column

**Reply by manojshah1976** [reply]
Thanks for this, good idea using a calculated field I initially went down the route of getting the worktime history then filtering based on the from and to status, sorting on the change date (asc/desc) same as your min max and then removing the duplicate tickets. Not sure what your view is with this approach.

**Reply by nbrown02** [reply]
I'd compare the two - see which gives the result mostly aligned with what you want :) Yours sounds like less rows of data though (always my preferred) !
